manifests/resource/server.pp

```diff
@@ -592,8 +594,29 @@
   if $ssl {
     # Access and error logs are named differently in ssl template
 
-    File <| title == $ssl_cert or path == $ssl_cert or title == $ssl_key or path == $ssl_key |>
-    -> concat::fragment { "${name_sanitized}-ssl-header":
+    if $ssl_key =~ Array {
+      $ssl_key.each | $key | {
+        File <| title == $key or path == $key |> {
+          before => Concat::Fragment["${name_sanitized}-ssl-header"],
+        }
+      }
+    } else {
+      File <| title == $ssl_key or path == $ssl_key |> {
+        before => Concat::Fragment["${name_sanitized}-ssl-header"],
+      }
+    }
```

Comment:
```suggestion
    $ssl_key.flatten.each |$key| {
      File <| title == $key or path == $key |> {
        before => Concat::Fragment["${name_sanitized}-ssl-header"],
      }
    }
```

Hmm, would something like this work to reduce code duplication?

Reply:
$ssl_key might be an Array, String or Boolean. I'm not sure, if this code is working like that.

Reply:
I'm not able to find a better way, as there is no way, I'm aware of, to use a variable for the collectors. E.g. `File <| $filter |>` isn't working. `File <| title = $array_var |>` isn't working either.

I'm open for suggestions here, as I don't like this part either.

Reply:
I think I've found a better way.